core/dbt/include/global_project/macros/materializations/models/incremental/on_schema_change.sql

```diff
@@ -128,6 +128,9 @@
 
           {% do exceptions.raise_compiler_error(fail_msg) %}
 
+        {% elif on_schema_change == 'full_refresh' %}
+          {{ return({"full_refresh": True}) }}
```

Comment:
What do you think about some additional debug-level logging here? Something like:
```
      {% do log("Full refreshing " ~ target_relation ~ " on account of schema change") %}
```